### PR TITLE
feat: Allow specifying the license for generated clients

### PR DESCRIPTION
**Pull request by FrankEssenberger**
Relates SAP/cloud-sdk-backlog#583.

we need an option to set the license for our generated clients. This PR adds the CLI option to set the license propert in the generate `package.json`.

### CODE REVIEW

File: packages/generator-common/src/file-writer/package-json.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  license?: string;
+}
+
+export function packageJsonBase(
+  options: PackageJsonOptions
+): Record<string, any> {
+  return {
+    name: options.npmPackageName,
+    version: options.version,
+    description: options.description,
```
Comment (florian-richter) [inline packages/generator-common/src/file-writer/package-json.ts:18]: This should have a sensible default value: `An [OData/OpenAPI] client for <packageName> generated by the SAP Cloud SDK v<SDK VERSION>`

File: packages/openapi-generator/src/options/options.ts [hunk 2]
```diff
@@ -89,6 +89,13 @@ export const generatorOptions = {
       'When enabled, a `package.json`, that specifies dependencies and scripts for transpilation and documentation generation is generated.',
     default: false
   },
+  licenseInPackageJson: {
+    string: true,
+    description:
+      "License to be used on the generated package.json. Only considered is 'packageJson' is enabled.",
```
Comment (florian-richter) [inline packages/openapi-generator/src/options/options.ts:95]:
```suggestion
      "License to be used on the generated package.json. Only considered if 'packageJson' is enabled.",
```

File: packages/generator-common/src/file-writer/package-json.ts (hunk 1 above)
Comment (FrankEssenberger) [reply packages/generator-common/src/file-writer/package-json.ts:18]: The two generators provide their description. What would be the value to have a default there, if it is not used under any circumstance?

**Review by florian-richter — APPROVED** [review]
LGTM

File: packages/generator-common/src/file-writer/package-json.ts (hunk 1 above)
Comment (FrankEssenberger) [reply packages/generator-common/src/file-writer/package-json.ts:18]: added issue to clean this up: https://github.com/SAP/cloud-sdk-backlog/issues/604

**Pull request merged**